### PR TITLE
remove trailing comma in JTD serialisation result, fixes #2001

### LINKED ISSUE #2001: Comma in serialization result
**What version of Ajv are you using? Does the issue happen if you use the latest version?**

**Version: 8.11.0**, **JTD**

**Ajv options object**

```javascript
{
  coerceTypes: 'array',
  useDefaults: true,
  removeAdditional: true,
  uriResolver: fastUri,
  allErrors: false
}
```

**JSON Schema**

```json
{
  "optionalProperties": {
    "test": {
      "type": "boolean"
    },
    "error": {
      "type": "string"
    }
  },
  "additionalProperties": false
}
```

**Sample data**

```json
{ "error": "Something went wrong" }
```

**Your code**

```javascript
const AjvJTD = require('ajv/dist/jtd')

const ajv = new AjvJTD({
  coerceTypes: 'array',
  useDefaults: true,
  removeAdditional: true,
  uriResolver: fastUri,
  allErrors: false
})
const serialize = ajv.compileSerializer(schema)
const result = serialize(object)
```

**Serialized data (NOTICE COMMA)** 
```
{,"error":"Something went wrong"}
```

**What results did you expect?**

````{"error":"Something went wrong"}````

**Are you going to resolve the issue?**
Yes, look at #2028

### PR DESCRIPTION
**Pull request by piliugin-anton**
fixes #2001, #2171, #2181

**What issue does this pull request resolve?**
https://github.com/ajv-validator/ajv/issues/2001
**What changes did you make?**
Added if condition to check whether property is first
**Is there anything that requires more attention while reviewing?**
Code correctness

### CODE REVIEW

**Comment by epoberezkin** [comment]
@piliugin-anton thank you!

It needs the test that fails without this change.

File: lib/compile/jtd/serialize.ts [hunk 1]
```diff
@@ -156,20 +156,20 @@ function serializeSchemaProperties(cxt: SerializeCxt, discriminator?: string): v
...
-      serializeProperty(key, optionalProperties[key], value)
+      serializeProperty(key, optionalProperties[key], value, true)
     )
   }
   if (schema.additionalProperties) {
     gen.forIn("key", data, (key) =>
-      gen.if(isAdditional(key, allProps), () =>
-        serializeKeyValue(cxt, key, {}, gen.let("first", first))
-      )
+      gen.if(isAdditional(key, allProps), () => serializeKeyValue(cxt, key, {}, firstProp))
```
Comment (epoberezkin) [inline lib/compile/jtd/serialize.ts:172]:
```suggestion
      gen.if(isAdditional(key, allProps), () =>
        serializeKeyValue(cxt, key, {}, firstProp))
```

File: lib/compile/jtd/serialize.ts [hunk 2]
```diff
@@ -189,9 +189,24 @@ function serializeSchemaProperties(cxt: SerializeCxt, discriminator?: string): v
...
 
-  function serializeProperty(key: string, propSchema: SchemaObject, value: Name): void {
-    if (first) first = false
-    else gen.add(N.json, str`,`)
+  function serializeProperty(
+    key: string,
+    propSchema: SchemaObject,
+    value: Name,
+    isOptional: boolean
+  ): void {
```
Comment (epoberezkin) [inline lib/compile/jtd/serialize.ts:197]:
```suggestion
  function serializeProperty(key: string, propSchema: SchemaObject, value: Name, isOptional: boolean): void {
```

**Comment by piliugin-anton** [comment]
@epoberezkin should I write a test or you will do it?

**Comment by epoberezkin** [comment]
@piliugin-anton it's great if you can - probably should be something similar to this one: https://github.com/ajv-validator/ajv/blob/master/spec/issues/1501_jtd_many_properties.spec.ts

File: lib/compile/jtd/serialize.ts [hunk 3]
```diff
@@ -189,9 +190,19 @@ function serializeSchemaProperties(cxt: SerializeCxt, discriminator?: string): v
     return gen.const("value", _`${data}${getProperty(key)}`)
   }
 
-  function serializeProperty(key: string, propSchema: SchemaObject, value: Name): void {
-    if (first) first = false
-    else gen.add(N.json, str`,`)
+  function serializeProperty(key: string, propSchema: SchemaObject, value: Name, isOptional: boolean): void {
```
Comment (epoberezkin) [inline lib/compile/jtd/serialize.ts]:
```suggestion
  function serializeProperty(
    key: string,
    propSchema: SchemaObject,
    value: Name,
    isOptional: boolean
  ): void {
```

**Comment by piliugin-anton** [comment]
@epoberezkin Okay, I will try. Will push the changes in an hour and ping you

**Comment by piliugin-anton** [comment]
@epoberezkin How do I write a test, when I already have a changes that returns correct result of serialization. Could you please assist me?

**Comment by piliugin-anton** [comment]
@epoberezkin Please check if test is correct

File: spec/issues/2001_jtd_only_optional_properties.spec.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import _Ajv from "../ajv_jtd"
+import * as assert from "assert"
+
+const PROP_COUNT = 3
+
+describe("schema with only optional properties", () => {
+  let ajv: InstanceType<typeof _Ajv>
```
Comment (epoberezkin) [inline spec/issues/2001_jtd_only_optional_properties.spec.ts:7]:
```suggestion
  let ajv: _Ajv
```

not 100% certain, but isn't it the same?

Comment (piliugin-anton) [reply spec/issues/2001_jtd_only_optional_properties.spec.ts:7]:
```TSError: ⨯ Unable to compile TypeScript:
spec/issues/2001_jtd_only_optional_properties.spec.ts:8:12 - error TS2749: '_Ajv' refers to a value, but is being used as a type here. Did you mean 'typeof _Ajv'?

7   let ajv: _Ajv
```

Comment (epoberezkin) [reply spec/issues/2001_jtd_only_optional_properties.spec.ts:7]: alright, ts is weird... :)

Comment (piliugin-anton) [reply spec/issues/2001_jtd_only_optional_properties.spec.ts:7]: I'm not a TypeScript expert, so I had to Google a solution, hope it's good

**Comment by epoberezkin** [comment]
amended in #2190

**Pull request closed without merging**